File: EduChemSuite.API/EduChemSuite.API/Models/ExamResponseModel.cs
namespace EduChemSuite.API.Models;

public class ExamResponseModel : BaseModel
{
    public required Guid ExamId { get; set; }
    public ExamModel? Exam { get; set; }
    public required Guid UserId { get; set; }
    public UserModel? User { get; set; }
    public required Guid QuestionId { get; set; }
    public QuestionModel? Question { get; set; }
    public Guid? ExamQuestionId { get; set; }
    public Guid? AnswerId { get; set; }
    public AnswerModel? Answer { get; set; }
    public String? ResponseText { get; set; }
    public String? ResponseImage { get; set; }
    public Guid? ImageTypeId { get; set; }
    public ImageTypeModel? ImageType { get; set; }
    public bool? IsCorrect { get; set; }
    public bool IsGraded { get; set; }
}

File: EduChemSuite.UI/UI/src/app/_models/ExamResponseModel.ts
import {BaseModel} from "./BaseModel";
import {UserModel} from "./UserModel";
import {QuestionModel} from "./QuestionModel";
import {AnswerModel} from "./AnswerModel";

export class ExamResponseModel extends BaseModel {
  examId?: string;
  userId?: string;
  user?: UserModel;
  questionId?: string;
  question?: QuestionModel;
  answerId?: string;
  answer?: AnswerModel;
  responseText?: string;
  responseImage?: string;
  imageTypeId?: string;
  isCorrect?: boolean | null;
  isGraded?: boolean;
}
